Add dbm lookup plugin

stoned opened the pull request:
##### SUMMARY
Pretty simple new lookup plugin to fetch values from DBM files.

##### ISSUE TYPE
 - Feature Pull Request


##### COMPONENT NAME
lookup plugin dbm

##### ANSIBLE VERSION
```
ansible 2.7.0.dev0 (dbm-lookup b6b20383ea) last updated 2018/06/12 22:45:50 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/seb/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /u/seb/ghq/github.com/stoned/ansible/lib/ansible
  executable location = /u/seb/ghq/github.com/stoned/ansible/bin/ansible
  python version = 2.7.15 (default, May 16 2018, 17:50:09) [GCC 8.1.1 20180502 (Red Hat 8.1.1-1)]
```


##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful. -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

stoned committed:
Add dbm lookup plugin

bcoca commented on lib/ansible/plugins/lookup/dbm.py:
```diff
@@ -0,0 +1,68 @@
+DOCUMENTATION = """
+    lookup: dbm
+    author: Stoned Elipot <stoned.elipot(at)gmail.com>
+    version_added: "2.6"
```
2.7 at this point

bcoca commented on lib/ansible/plugins/lookup/dbm.py:
```diff
@@ -0,0 +1,68 @@
+      - values of keys.
+    type: list
+"""
+import anydbm
```
in py3 this is dbm, so try/except

stoned committed:
be Python 3 friendly; adjust version for future ansible release

mattclay commented:
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/70120/32/tests

stoned committed:
dbm lookup test tweaks

- install required dependency, python-gdbm, on openSUSE Leap
- make the test DBM file creation Python 2/3 aware

stoned committed:
fix whitespace !

mattclay commented:
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/72512/35/tests

mattclay commented:
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/73460/35/tests

stoned committed:
Add an encoding option to dbm lookup

Decode value from dbm lookup plugin using the encoding specified in the
optional 'encoding' argument which defaults to utf-8.

stoned committed:
Explicitly specify encoding of data in dbm lookup test database

The dbm lookup now defaults to utf-8 when decoding keys' value, and tests
do not override this default, so (really) make sure the tests data is
in uft-8.

stoned committed:
Placate pep8 sanity test

resmo committed:
minors

resmo commented:
build failed, this needs investigation before we could merge.

Akasurde commented:
Hi @stoned, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.general<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>
